oci_client.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"fmt"
+	"github.com/oracle/oci-go-sdk/common"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+)
+
+// OciClient stores the client and configuration details for making API requests to OCI Identity Service
+type OciClient struct {
+	common.BaseClient
+	config *common.ConfigurationProvider
+}
+
+// These constants store information related to signing the http request
+const (
+	// requestHeaderDate The key for passing a header to indicate Date
+	requestHeaderDate = "Date"
+
+	// requestHeaderUserAgent The key for passing a header to indicate User Agent
+	requestHeaderUserAgent = "User-Agent"
+
+	defaultScheme = "https"
+)
+
+// NewIdentityClientWithConfigurationProvider Creates a new default Identity client with the given configuration provider.
+// the configuration provider will be used for the default signer as well as reading the region
+func NewOciClientWithConfigurationProvider(configProvider common.ConfigurationProvider) (client OciClient, err error) {
+	baseClient, err := common.NewClientWithConfig(configProvider)
+	if err != nil {
+		return
+	}
+
+	client = OciClient{BaseClient: baseClient}
+	client.BasePath = ""
+	err = client.setConfigurationProvider(configProvider)
+	return
+}
+
+// SetRegion overrides the region of this client.
+func (client *OciClient) SetHost(host string) {
+	client.Host = host
+}
+
+// SetConfigurationProvider sets the configuration provider including the region, returns an error if is not valid
+func (client *OciClient) setConfigurationProvider(configProvider common.ConfigurationProvider) error {
+	if ok, err := common.IsConfigurationProviderValid(configProvider); !ok {
+		return err
+	}
+
+	// Error has been checked already
+	client.config = &configProvider
+	return nil
+}
+
+// ConstructLoginRequest takes in a path and returns a signed http request
+func (client OciClient) ConstructLoginRequest(path string) (request http.Request, err error) {
+	httpRequest, err := common.MakeDefaultHTTPRequestWithTaggedStruct(http.MethodGet, path, request)
+	if err != nil {
+		return
+	}
+
+	err = client.prepareRequest(&httpRequest)
+	if err != nil {
+		return
+	}
+
+	err = client.Signer.Sign(&httpRequest)
+	if err != nil {
+		return
+	}
+
+	request = httpRequest
+	err = nil
+	return
+}
+
+// prepareRequest takes in a http request and adds the required information for signing it
+func (client *OciClient) prepareRequest(request *http.Request) (err error) {
+	if client.UserAgent == "" {
+		return fmt.Errorf("user agent can not be blank")
```

Comment:
```suggestion
		return errors.New("user agent can not be blank")
```
(since there are no parameters, no need to parse)